tests/test_pytorch_models.py

```diff
@@ -79,129 +81,60 @@ def forward(self, x):
-            )
-
-            self.assertEqual(EPOCH, model_out_dict["epoch"])
-            self.assertEqual(LOSS, model_out_dict["loss"])
-
-            # Check model parameters
-            for key_item_1, key_item_2 in zip(
-                net.state_dict().items(), loaded_net.state_dict().items()
-            ):
-                self.assertTrue(torch.equal(key_item_1[1], key_item_2[1]))
-
-            # Check optimizer parameters
-            optimizer_state = optimizer.state
-            loaded_optimizer_state = loaded_optimizer.state
-
-            for key_item_1, key_item_2 in zip(
-                optimizer_state.items(), loaded_optimizer_state.items()
-            ):
-                self.assertTrue(torch.equal(key_item_1[1]["sum"], key_item_2[1]["sum"]))
-
-    def test_save_and_load_nn(self):
-        with tempfile.TemporaryDirectory() as tiledb_array:
-            net = Net()
-            optimizer = optim.Adam(net.parameters(), lr=0.001)
-            EPOCH = 5
-            LOSS = 0.4
-
-            tiledb_obj = PyTorchTileDB(uri=tiledb_array)
-            tiledb_obj.save(
-                model_info={
-                    "epoch": EPOCH,
-                    "model_state_dict": net.state_dict(),
-                    "optimizer_state_dict": optimizer.state_dict(),
-                    "loss": LOSS,
-                }
-            )
-
-            loaded_net = Net()
-            loaded_optimizer = optim.Adam(loaded_net.parameters(), lr=0.001)
-            model_out_dict = tiledb_obj.load(
-                model=loaded_net, optimizer=loaded_optimizer
-            )
-
-            self.assertEqual(EPOCH, model_out_dict["epoch"])
-            self.assertEqual(LOSS, model_out_dict["loss"])
-
-            # Check model parameters
-            for key_item_1, key_item_2 in zip(
-                net.state_dict().items(), loaded_net.state_dict().items()
-            ):
-                self.assertTrue(torch.equal(key_item_1[1], key_item_2[1]))
-
-            # Check optimizer parameters
-            optimizer_state = optimizer.state
-            loaded_optimizer_state = loaded_optimizer.state
-
-            for key_item_1, key_item_2 in zip(
-                optimizer_state.items(), loaded_optimizer_state.items()
-            ):
-                self.assertTrue(torch.equal(key_item_1[1]["sum"], key_item_2[1]["sum"]))
-
-
-if __name__ == "__main__":
-    unittest.main()
+@pytest.fixture
+def temp_uri():
+    """
+    Returns a temporary directory instance
+    """
+    return tempfile.TemporaryDirectory()
+
+
+@pytest.mark.parametrize(
+    "optimizer",
+    [
+        getattr(optimizers, name)
+        for name, obj in inspect.getmembers(optimizers)
+        if inspect.isclass(obj) and name != "Optimizer"
+    ],
+)
+@pytest.mark.parametrize(
+    "net",
+    [
+        getattr(sys.modules[__name__], name)
+        for name, obj in inspect.getmembers(sys.modules[__name__])
+        if inspect.isclass(obj) and obj.__module__ == __name__
+    ],
```

Comment:
Yes it is a little bit hacky. I designed it this way in case we want to expand in the future with extra network definitions. Cause tbh the current ones are a small subset and basic testing scenarios.